lib/monitoring/monitoringClientInterface.js

```diff
@@ -0,0 +1,128 @@
+'use strict';
+
+const assert = require('assert')
+const client = require('prom-client');
+
+/**
+ * Wrapper for Prometheus client - https://github.com/siimon/prom-client
+ */
+
+var references = {};
+var namespace = '';
+
+module.exports = {
+  /**
+   * @param {string} ns - Sets a namespace for all the metrics
+   */
+  setNamespace: function(ns) {
+      assert(typeof ns === 'string');
+      namespace = ns;
+  },
+
+  /** getMetrics
+   *  Gets all the metrics in the form of a string for prometheus server
+   */
+  getMetrics: function() {
+    return client.register.metrics();
+  },
+
+  /** 
+   * Get metrics by its name
+   * @param {string} metricsName - name of the metrics
+   */
+  get: function (metricsName) {
+    return references[metricsName];
+  },
+  
+  /**
+   * Counters go up, and reset when the process restarts.
+   * @param {string} metricsName - name of the metrics
+   * @param {string} metricsHelp - description of metrics
+   * @param {string} metricsLabels - Static labels applied to every metric.
+   * Optional Parameter - metricsLabels 
+   */
+  createCounter: function (metricsName, metricsHelp, metricsLabels) {
+    if (metricsLabels === undefined) {
+        metricsLabels = [];
+    }
+    references[metricsName] = new client.Counter({
+        name: metricsName,
```

Comment:
Even though I reject the thought of using 'namespaces' (see comment above), I guess this lacks the corresponding code to modify the given `metricsName`, as done for other metric types.

Reply:
Removed namespace